src/jobflow_remote/jobs/jobcontroller.py

```diff
@@ -2042,8 +2046,21 @@ def get_flow_job_aggreg(
         if query:
             pipeline.append({"$match": query})
 
-        if projection:
-            pipeline.append({"$project": projection})
+        if projection_flow:
+            pipeline.append({"$project": projection_flow})
+
+        if projection_job:
+            # insert the pipeline for the projection of the Job fields
+            # to reduce the impact of the size of the documents.
+            # This can help reducing the size of the fetched documents and
+            # avoid exceeding the maximum size allowed. Adding the projection
+            # in the general pipeline does not have the same effect.
+            pipeline[0]["$lookup"]["pipeline"] = [{"$project": projection_job}]
+            # if the additional projection is set, the keys need to be specified
+            # in that part of the pipeline as well.
+            if projection_flow:
+                for k in projection_job:
+                    pipeline[-1]["$project"][f"jobs_list.{k}"] = 1
```

Comment:
I am not sure what is the suggestion here. Making two different queries in python and match the results? This is probably not very efficient. Otherwise this behaves roughly like a `JOIN` in SQL, so it first needs to perform the matches in order to make the filter. I don't think the order can be altered here.

Reply:
no, I meant to change the pipeline order. But again, I'm not completely sure about this :)